### PR TITLE
Implement BootStrap Status Infinite Retries 

### PR DESCRIPTION
**Pull request by DavidjohnBlodgett**
This PR addresses a case were, if we are performing tasks such as flashing BIOS, we would  currently stop attempting to report a nodes success to a server (in the event that the server is down for some reason), after a set amount of retries. In such an event, we could have a disconnection between current and expected state on that node. This implements a fix that will have the node attempt to report its success indefinitely, until the server responds.

@RackHD/corecommitters

### CODE REVIEW

File: data/templates/bootstrap.js [hunk 1]
```diff
@@ -54,9 +53,13 @@ function eachSeries(arr, iterator, callback) {
  * @private
  * @param data
  * @param timeout
- * @param retries
  */
-function updateTasks(data, timeout, retries) {
+function updateTasks(data, timeout) {
+    // value to check succes case.
```
Comment (benbp) [inline data/templates/bootstrap.js]: These comments are redundant, it's pretty clear from the code what is happening already.

**Comment by benbp** [comment]
👍

**Comment by benbp** [comment]
Would be nice if you could change the commit string to be a little more descriptive.

File: data/templates/bootstrap.js (hunk 1 above)
Comment (DavidjohnBlodgett) [inline data/templates/bootstrap.js]: understood.

**Comment by DavidjohnBlodgett** [comment]
@benbp oh ok, sorry about that I'll fix it up post haste.

**Comment by benbp** [comment]
@DavidjohnBlodgett no worries! I'm just being picky :)

**Comment by DavidjohnBlodgett** [comment]
test this please

**Comment by tldavies** [comment]
test this please

**Comment by tldavies** [comment]
👍 when jenkins passes

**Comment by tldavies** [comment]
test this please

**Comment by tldavies** [comment]
test this please

**Comment by yyscamper** [comment]
:+1:

**Comment by brianparry** [comment]
test this please

**Comment by DavidjohnBlodgett** [comment]
@jlongever Hey Joel, I'm sorry to have to bug you... but I'm hoping I could tap your knowledge to assist in finding the root cause behind Jenkins timing out, I wouldn't put it beyond the realm of possibility that me introducing an action that no longer times out or has a max retry could mess with something, but it isn't jumping out to me that a smoke test is failing or anything... Just wanted a 3rd set of eyes on it, if I could... thanks!

**Comment by brianparry** [comment]
@DavidjohnBlodgett We've had some issues with our CI system over the past few days.  The timeouts on this PR are probably not caused by your change.  Jenkins, test this please

**Comment by DavidjohnBlodgett** [comment]
@brianparry oh well thats comforting in a way lol... understood, I'll stand by until more info is available.

**Comment by brianparry** [comment]
@DavidjohnBlodgett This looks like a real failure.  I think the discovery graph is getting stuck.

**Comment by DavidjohnBlodgett** [comment]
@brianparry @jlongever @benbp Hi guys, so I've been trying to wrap my brain around what's the issue here, and can you guys help with some insight to what is happening here? https://github.com/RackHD/RackHD/blob/master/test/tests/api/v1_1/nodes_tests.py#L120

It feels to me like this code maybe actually waiting for the "else" case found here: https://github.com/RackHD/on-http/pull/349/files#diff-fa828f1c5e9bca4ba39e54d7c31b1c2bR98

does that seem to jive with what you see? I am assuming this is failing our tests because we are expecting the update tasks function to only retry a limited amount of time, then bail out to pickup the next task... and this code would keep it attempting to respond indefinitely.

Just wanted to see if my logic is sound...

**Comment by jlongever** [comment]
@DavidjohnBlodgett That discovery test spawns a thread that listens on AMQP for a `graph.finished` event for each node that completed discovery graph.   The `wait_for_compeletion` is waiting for each worker thread to complete,  it will timeout after 20 minutes.   I think your assumption is correct,  the discovery graph never triggered a `graph.finished` event because discovery never completed?  Looking at the log here http://rackhdci.lss.emc.com/job/on-http/1747/artifact/build/vagrant.log I see `Graph.Discovery` task starting,  but doesn't seem like it runs through all of the catalog tasks.

**Comment by tldavies** [comment]
@jlongever @DavidjohnBlodgett So what would be a good way to change the test? As it used to complete cause it would timeout after 5 tries and move on the next step. Now we want infinity until we get a response from the monorail server. Which it seems is why it is timing out as it still wanting for the response?

**Comment by DavidjohnBlodgett** [comment]
@RackHD/corecommitters bug fixed, commit squashed. This should be good to go.

**Comment by tldavies** [comment]
test this please

File: data/templates/bootstrap.js [hunk 2]
```diff
@@ -79,33 +78,22 @@ function updateTasks(data, timeout, retries) {
...
+                retries = 1;
+            }else {
+                retries = retries + 1;
+            }
             console.log("Retrying Update Tasks Attempt #" + retries);
 
             setTimeout(function () {
-                updateTasks(data, timeout, retries);
+                updateTasks(data, timeout, retry, retries);
             }, timeout * retries);
```
Comment (zyoung51) [inline data/templates/bootstrap.js]: @DavidjohnBlodgett Now that you are allowing for an infinite number of retries should you still scale this up by the number of retries without some kind of upper limit?

Comment (DavidjohnBlodgett) [inline data/templates/bootstrap.js]: @zyoung51 I agree with your suggestion, and have added a ceiling of about 5mins. commit just posted.

**Comment by tldavies** [comment]
test this please

**Comment by DavidjohnBlodgett** [comment]
@RackHD/corecommitters will need to squash again... but just looking for thumbs up~ I believe I have addressed everyone's concerns.

**Comment by DavidjohnBlodgett** [comment]
test this please

**Pull request merged**